tasks/base_tasks.py

```diff
@@ -1592,3 +1592,127 @@ def collect_meta_wrappers(test_module=None, test_all=True):
             yield t, params
             if not test_all:
                 break
+
+
+class CreateRepoTable(Task):
+    schema = 'repo'
+    table = 'filerepo'
+
+    def run(self):
+        session = current_session()
+
+        try:
+            create_schema = '''
+                            CREATE SCHEMA IF NOT EXISTS "{schema}"
+                            '''.format(schema=self.output().schema)
+            session.execute(create_schema)
+
+            create_table = '''
+                            CREATE TABLE IF NOT EXISTS "{schema}".{table} (
+                                id TEXT,
+                                version NUMERIC,
+                                checksum TEXT,
+                                url TEXT,
+                                path TEXT,
+                                added TIMESTAMP,
+                                PRIMARY KEY(id, version)
+                            );
+                            '''.format(schema=self.output().schema,
+                                       table=self.output().tablename)
+            session.execute(create_table)
+            session.commit()
+        except:
+            session.rollback()
+
+    def output(self):
+        return PostgresTarget(schema=self.schema, tablename=self.table, non_empty=False)
+
+
+def base_downloader(url, output_path):
+    urllib.request.urlretrieve(url, output_path)
+
+
+class RepoFile(Task):
+    resource_id = Parameter()
+    url = Parameter()
+    version = IntParameter()
+    downloader = Parameter(default=base_downloader)
```

Comment:
Is this really needed? Parameters in luigi are not like arguments in the constructor, they define the instance name of the class so if this is needed maybe we need to add the [significant](http://luigi.readthedocs.io/en/stable/parameters.html#insignificant-parameters) property